fixpatches/src/commonMain/kotlin/com/saveourtool/sarifutils/adapter/SarifFixAdapter.kt

```diff
@@ -68,21 +69,34 @@ class SarifFixAdapter(
      * @return list of files with applied fixes
      */
     fun process(): List<Path> {
+        if (targetFiles.isEmpty()) {
+            log.warn { "The list of target files is empty." }
+            return emptyList()
+        }
+
         val sarifSchema210: SarifSchema210 = Json.decodeFromString(
             readFile(sarifFile)
         )
         // A run object describes a single run of an analysis tool and contains the output of that run.
         val processedFiles = sarifSchema210.runs.asSequence().flatMapIndexed { index, run ->
             val runReplacements: List<RuleReplacements> = extractFixObjects(run)
             if (runReplacements.isEmpty()) {
-                log.warn { "The run #$index doesn't have any `fix object` section!" }
-                emptyList()
+                log.warn { "Run #${index + 1} doesn't have any `fix object` section!" }
+                emptySequence()
             } else {
                 val groupedReplacements = groupReplacementsByFiles(runReplacements)
                 // if there are several fixes by different rules for the same region within one file, take only first of them
                 // and reverse the order of replacements for each file
                 val filteredRuleReplacements = filterRuleReplacements(groupedReplacements)
-                applyReplacementsToFiles(filteredRuleReplacements, targetFiles)
+
+                when {
+                    filteredRuleReplacements.isEmpty() -> {
```

Comment:
According to the JSON schema, it couldn't (the minimum size is 1).

Technically, it could.
And there's a separate test (with SARIF input data) which covers this case.